src/gui/render.cpp

```diff
@@ -433,10 +444,29 @@ static void RENDER_Reset( void ) {
 		}
 	}
 /* Setup the scaler variables */
-#if C_OPENGL
-	GFX_SetShader(render.shader_src);
-#endif
-	gfx_flags=GFX_SetSize(width,height,gfx_flags,gfx_scalew,gfx_scaleh,&RENDER_CallBack);
+	double par;  /* the pixel aspect ratio of the source pixel array */
+	double parn; /* normalised PAR, always > 1.0 */
+	par = (double)width / height / 4 * 3; /* MS-DOS screen is always 4:3 */
```

Comment:
*dreamer*:

> I suggest renaming these variables

The names you suggest are quite long and make expressions exceedingly cumbersome. Since the algorithm nowhere involves such concept as the *destination PAR*, I suggest to drop the `src` prefix. It is not justified because there is no `dst_par`. The *destination PAR* is always unity on LCD displays.

*PAR* is the [standard](https://en.wikipedia.org/wiki/Pixel_aspect_ratio) abbreviation for pixel aspect ratio, so I propose to keep the abbreviated forms: `par` and `par_norm`.

If you insist on removing the `par` root, how about `pixel_asp` for *pixel aspect*?

> But I am not sure about 1.0 part—shouldn't normalised ratio sometimes be == 1.0?

Perhaps I misapplied the term *normalised*. Anyway, your review has shown that `parn` is a rudiment of an unfinished feature that I have removed altogether for now.

> (if it can be, then we have a problem with storing this value as double).

What is the problem? A `double` can store integer numbers with absolute precision.

> Also, not sure where `dblh`, `dblw` came from in this code, but these variable names are too short.

This notation for double-width and double-height flags is used throughout *DOSBox* code. I should not like to change it everywhere withing this *PR*.

> One of good rules of thumb to follow when naming variables: length/descriptiveness should be proportional to the scope of the variable

I agree.

> when scope reaches to different file, variable is a global one, etc - the longer name is needed, e.g. `use_double_width`

I hope simply `double_width` is enough for a `bool` variable.